arrow-core/src/main/kotlin/arrow/core/extensions/andthen.kt

```diff
@@ -44,45 +40,45 @@ interface AndThenMonoid<A, B> : Monoid<AndThen<A, B>>, AndThenSemigroup<A, B> {
 }
 
 @extension
-interface AndThenFunctor<X> : Functor<AndThenPartialOf<X>> {
-  override fun <A, B> AndThenOf<X, A>.map(f: (A) -> B): AndThen<X, B> =
-    fix().map(f)
+interface AndThenFunctor : Functor<ForAndThen> {
+  override fun <A, B> AndThenPartialOf<A>.map(f: (A) -> B): AndThenPartialOf<B> =
```

Comment:
I believe this may not be correct. We want to return concrete instances from the functions, so `AndThen<X, B>` is what's expected as a return.

Reply:
This applies to the others too. I ned to check the rest of the diff.

Reply:
Good point, i'll take another look then, thanks mate ❤️